aa-user/register_application.go

```diff
@@ -8,39 +8,37 @@ import (
 	"github.com/PretendoNetwork/nex-protocols-go/globals"
 )
 
-// RegisterApplication sets the RegisterApplication handler function
-func (protocol *Protocol) RegisterApplication(handler func(err error, packet nex.PacketInterface, callID uint32, titleID uint64) uint32) {
-	protocol.registerApplicationHandler = handler
-}
-
 func (protocol *Protocol) handleRegisterApplication(packet nex.PacketInterface) {
 	var errorCode uint32
 
-	if protocol.registerApplicationHandler == nil {
+	if protocol.RegisterApplication == nil {
 		globals.Logger.Warning("AAUser::RegisterApplication not implemented")
 		go globals.RespondError(packet, ProtocolID, nex.Errors.Core.NotImplemented)
 		return
 	}
 
-	request := packet.RMCRequest()
+	request := packet.RMCMessage()
 
-	callID := request.CallID()
-	parameters := request.Parameters()
+	callID := request.CallID
+	parameters := request.Parameters
 
 	parametersStream := nex.NewStreamIn(parameters, protocol.Server)
 
 	titleID, err := parametersStream.ReadUInt64LE()
 	if err != nil {
-		errorCode = protocol.registerApplicationHandler(fmt.Errorf("Failed to read titleID from parameters. %s", err.Error()), packet, callID, 0)
+		_, errorCode = protocol.RegisterApplication(fmt.Errorf("Failed to read titleID from parameters. %s", err.Error()), packet, callID, 0)
```

Comment:
I don't think we need to send the whole packet to the method handlers anymore, unless I'm missing something that could be useful?

Reply:
Actually the common lib does send notifications on it's own. So having access to the original packet there would make it easier to do things like not assume the servers PRUDP version

Reply:
Oh yeah right! We can leave the packet as argument then